packages/block-library/src/template-part/edit/index.js

```diff
@@ -168,6 +173,7 @@ export default function TemplatePartEdit( {
 					getSettings().onNavigateToEntityRecord,
 				title: entityRecord?.title,
 				canUserEdit: !! _canUserEdit,
+				isNavigationMode: _isNavigationMode(),
```

Comment:
Suggestion (non-blocking): Should we return `onNavigateToEntityRecord` as undefined when the block editor isn't in navigation mode? This value is only used with that method.